src/rpc/handlers/AMMInfo.cpp

```diff
@@ -0,0 +1,245 @@
+            boost::json::object vote;
+            vote[JS(account)] = to_string(voteEntry.getAccountID(sfAccount));
+            vote[JS(trading_fee)] = voteEntry[sfTradingFee];
+            vote[JS(vote_weight)] = voteEntry[sfVoteWeight];
+
+            response.voteSlots.push_back(std::move(vote));
+        }
+    }
+
+    if (amm.isFieldPresent(sfAuctionSlot)) {
+        auto const& auctionSlot = amm.peekAtField(sfAuctionSlot).downcast<STObject>();
+        if (auctionSlot.isFieldPresent(sfAccount)) {
+            boost::json::object auction;
+            auto const timeSlot = ammAuctionTimeSlot(lgrInfo.parentCloseTime.time_since_epoch().count(), auctionSlot);
+
+            auction[JS(time_interval)] = timeSlot ? *timeSlot : AUCTION_SLOT_TIME_INTERVALS;
+            auction[JS(price)] = toBoostJson(auctionSlot[sfPrice].getJson(JsonOptions::none));
+            auction[JS(discounted_fee)] = auctionSlot[sfDiscountedFee];
+            auction[JS(account)] = to_string(auctionSlot.getAccountID(sfAccount));
+            auction[JS(expiration)] = toIso8601(NetClock::time_point{NetClock::duration{auctionSlot[sfExpiration]}});
+
+            if (auctionSlot.isFieldPresent(sfAuthAccounts)) {
+                boost::json::array auth;
+                for (auto const& acct : auctionSlot.getFieldArray(sfAuthAccounts)) {
+                    boost::json::object accountData;
+                    accountData[JS(account)] = to_string(acct.getAccountID(sfAccount));
+                    auth.push_back(accountData);
+                }
+
+                auction[JS(auth_accounts)] = std::move(auth);
+            }
+
+            response.auctionSlot = std::move(auction);
+        }
+    }
+
+    if (!isXRP(asset1Balance)) {
+        response.asset1Frozen = isFrozen(
+            *sharedPtrBackend_, lgrInfo.seq, ammAccountID, amm[sfAsset].currency, amm[sfAsset].account, ctx.yield
+        );
+    }
+    if (!isXRP(asset2Balance)) {
+        response.asset2Frozen = isFrozen(
+            *sharedPtrBackend_, lgrInfo.seq, ammAccountID, amm[sfAsset2].currency, amm[sfAsset2].account, ctx.yield
+        );
+    }
+
+    return response;
+}
+
+void
+tag_invoke(boost::json::value_from_tag, boost::json::value& jv, AMMInfoHandler::Output const& output)
+{
+    boost::json::object amm = {
+        {JS(lp_token), output.lpToken},
+        {JS(amount), output.amount1},
+        {JS(amount2), output.amount2},
+        {JS(account), output.ammAccount},
+        {JS(trading_fee), output.tradingFee},
+    };
+
+    if (output.auctionSlot != nullptr)
+        amm[JS(auction_slot)] = output.auctionSlot;
+
+    if (not output.voteSlots.empty())
+        amm[JS(vote_slots)] = output.voteSlots;
+
+    if (output.asset1Frozen)
+        amm[JS(asset_frozen)] = *output.asset1Frozen;
+
+    if (output.asset2Frozen)
+        amm[JS(asset2_frozen)] = *output.asset2Frozen;
+
+    jv = {
+        {JS(amm), amm},
+        {JS(ledger_current_index), output.ledgerIndex},
```

Comment:
ledger_current_index is not for Clio, rippled returns ledger_current_index for not validated ledger. For Clio, validated will be always true, thus only ledger_index.

Reply:
Ok, i will update it back then. The docs are not specific about the difference and since `rippled` returns `ledger_current_index` i thought we should too because Mounika will definitely come to us with questions later 👍